backend/src/repositories/readStockRepository.ts

```diff
@@ -3,15 +3,29 @@ import {FieldPacket, PoolConnection, RowDataPacket} from "mysql2/promise";
 export const readAllStocks = async (
     connection: PoolConnection
 ) => {
-    try {
-        const stocks = (await connection.query("SELECT * FROM stocks")) as [
-            RowDataPacket[],
-            FieldPacket[]
-        ];
-        return stocks[0]
-    }
-    catch(err :any) {
-        console.log('Error occured when retrieving all stocks', err);
-        throw err;
-    }
-}
\ No newline at end of file
+    const stocks = (await connection.query("SELECT * FROM stocks")) as [
+        RowDataPacket[],
+        FieldPacket[]
+    ];
+    return stocks[0]
```

Comment:
suggestion: tester en end to end, lorsqu'il n'y aucune ligne dans stocks que ça ne fasse pas tomber l'application